packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -1177,7 +1177,7 @@ class Isolate extends ServiceObjectOwner {
       final Map<String, dynamic> response = await invokeRpcRaw('_reloadSources', params: arguments);
       return response;
     } on rpc.RpcException catch (e) {
-      return Future<Map<String, dynamic>>.error(<String, dynamic>{
```

Comment:
This was the problem. I'm not sure why the `on Map` catch wasn't catching this error.

Reply:
The `Future.error()` probably skips the try/catch and gets thrown to the Zone onError. If this had been a `throw <String, dynamic>{...}`, then it would have been handled by the `on Map`.

Reply:
Does the handling of what was supposed to be exception now is return value need to be changed?

Reply:
I've confirmed that returning this error goes through our existing code path for handling reload errors.